src/renderer/src/hooks/stores/mapeoDeviceStore.ts

```diff
@@ -0,0 +1,103 @@
+import { create } from 'zustand'
+
+type DeviceType = 'mobile' | 'desktop'
+
+export type Role = 'coordinator' | 'participant'
+
+type Device = {
+  name: string
+  deviceId: string
+  deviceType: DeviceType
+  pastDeviceNames?: string[]
+}
+
+type MemberDevice = {
+  role: Role
+  dateAdded: Date
+} & Device
+
+type MapeoDeviceState = {
+  nonMemberDevices: Record<Device['deviceId'], Device>
+  memberDevices: Record<MemberDevice['deviceId'], MemberDevice>
+  actions: {
+    addDeviceToProject: (deviceId: string, role: Role) => void
+    removeDeviceFromProject: (deviceId: string) => void
+    changeDeviceRole: (deviceId: string, newRole: Role) => void
+  }
+}
+
+export const useMapeoDeviceStore = create<MapeoDeviceState>()((set) => ({
+  nonMemberDevices: createRandomDevices(),
+  memberDevices: {},
+  actions: {
+    addDeviceToProject: (deviceId, role) =>
+      set((state) => {
+        if (state.memberDevices[deviceId]) {
+          throw new Error(`Device is already part of the project`)
+        }
+
+        const deviceToBeAdded = state.nonMemberDevices[deviceId]
+
+        if (!deviceToBeAdded) {
+          throw new Error(`Device does not exist`)
+        }
+
+        const { [deviceId]: key, ...rest } = state.nonMemberDevices
+
+        return {
+          nonMemberDevices: rest,
+          memberDevices: {
+            ...state.memberDevices,
+            [deviceId]: {
+              ...deviceToBeAdded,
+              role,
+              dateAdded: new Date(),
+            },
+          },
+        }
+      }),
+    removeDeviceFromProject: (deviceId) =>
+      set((state) => {
+        const deviceToBeRemoved = state.memberDevices[deviceId]
+        if (!deviceToBeRemoved) {
+          throw new Error(`Device is not part of the project`)
+        }
+        const { [deviceId]: key, ...newMemberDevices } = state.memberDevices
```

Comment:
similar nonblocking suggestion:

```suggestion
        const { [deviceId]: deviceToBeRemoved, ...newMemberDevices }  = state.memberDevices
        
        if (!deviceToBeRemoved) {
          throw new Error(`Device is not part of the project`)
        }
```

Reply:
https://github.com/digidem/mapeo-desktop-shell/pull/40/commits/9d76998e7e969ecdb734c309a9249eed7a8b74c6